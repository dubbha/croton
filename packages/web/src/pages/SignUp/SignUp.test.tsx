import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

jest.mock('components', () => ({
  Container: ({ children }) => <div className="container">{children}</div>,
  Header: () => <div>Header</div>,
  Footer: () => <div>Footer</div>,
  SignUpForm: ({ onSubmit }) => <button onClick={onSubmit} data-testid="submitButton" className="signUpForm" />,
}));

describe('pages/SignUp', () => {
  it('should render successfully', () => {
    jest.isolateModules(() => {
      jest.doMock('react-redux', () => ({
        useDispatch: () => jest.fn(),
        useSelector: () => false,
      }));

      const { SignUp } = require('./SignUp');

      const { container } = render(<SignUp />, { wrapper: MemoryRouter });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should dispatch on submit', () => {
    const fn = jest.fn();
    jest.isolateModules(() => {
      jest.doMock('react-redux', () => ({
        useDispatch: () => fn,
        useSelector: () => false,
      }));

      jest.doMock('components/SignUpForm', () => {
        const React = require('react');
        return {
          SignUpForm: ({ onSubmit }) => (
            <button onClick={onSubmit} data-testid="submitButton" />
          ),
        };
      });

      const { SignUp } = require('./SignUp');

      const { getByTestId } = render(<SignUp />, { wrapper: MemoryRouter });
      const submitButton = getByTestId('submitButton');

      fireEvent.click(submitButton);
      expect(fn).toBeCalled();
    });
  });
});
