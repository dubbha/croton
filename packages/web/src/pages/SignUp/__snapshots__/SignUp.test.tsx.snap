// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/SignUp should render successfully 1`] = `
<div
  class="container"
>
  <header>
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="navbar-brand"
          href="/#intro"
        >
          <img
            alt="logo"
            class="d-inline-block align-top"
            height="60"
            src="logo.svg"
            width="60"
          />
          <img
            alt="Croton. Care. Free."
            src="croton.png"
          />
        </a>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/#features"
          >
            Features
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/#about"
          >
            About
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/#services"
          >
            Services
          </a>
        </div>
      </div>
      <div
        class="navbar-nav"
      >
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/signin"
          >
            Sign In
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </div>
    </nav>
  </header>
  <form
    class=""
  >
    <div
      style="height: 50px; margin-bottom: 16px;"
    />
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formFirstName"
      >
        First Name
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__firstName"
        id="formFirstName"
        name="firstName"
        placeholder="Enter First Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formLastName"
      >
        Last Name
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__lastName"
        id="formLastName"
        name="lastName"
        placeholder="Enter Last Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formEmail"
      >
        Email address
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__email"
        id="formEmail"
        name="email"
        placeholder="Enter Email"
        type="email"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formPassword"
      >
        Password
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__password"
        id="formPassword"
        name="password"
        placeholder="Enter password"
        type="password"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="formPasswordMatch"
      >
        Repeat Password
      </label>
      <input
        class="form-control"
        data-testid="signUpForm__passwordMatch"
        id="formPasswordMatch"
        name="passwordMatch"
        placeholder="Repeat password"
        type="password"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
      data-testid="submitButton"
      disabled=""
      type="submit"
    >
      <div
        class="spinner-container"
      />
      <span>
        Sign Up
      </span>
    </button>
  </form>
  <div
    class="signup-login"
  >
    Already have an account? 
    <a
      href="/signin"
    >
      Sign In
    </a>
  </div>
  <footer>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="copyright"
          >
            Â© 
            2020
             
            <a
              href="https://github.com/dubbha/croton"
            >
              Croton
            </a>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="justify-content-end nav"
          >
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="https://opensource.org/licenses/MIT"
                href="https://opensource.org/licenses/MIT"
                target="_blank"
              >
                MIT License
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                href="#"
                role="button"
                target="_blank"
              >
                Contact Us
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
